File: day1/javascript/d01a.js
const fs = require('fs')
const readline = require('readline')

const fileLines = []

const fileInterface = readline.createInterface({
    input: fs.createReadStream('../../inputs/d01.txt')
})

fileInterface.on('line', line => {
    fileLines.push(+line)
})

fileInterface.on('close', () => {
    const answer = fileLines
        .slice(1).reduce((previous, current) => Object.assign(previous, {
            prev: current,
            count: previous.count + (current > previous.prev ? 1 : 0)
        }), {
            prev: fileLines[0],
            count: 0
        })
    console.log(answer)
})

File: day1/python/d01a.py
with open('../../inputs/d01.txt', 'r') as file:
    data = [int(line) for line in file.readlines()]

prev = data[0]
count = 0

for line in data[1:]:
    if line > prev:
        count += 1
    prev = line

print(count)
